x-pack/plugins/apm/public/context/url_params_context/url_params_context.tsx

```diff
@@ -75,12 +79,13 @@ const UrlParamsProvider: React.ComponentClass<{}> = withRouter(
 
     refUrlParams.current = urlParams;
 
+    const incrementRangeId = () => setRangeId((prevRangeId) => prevRangeId + 1);
+
     const refreshTimeRange = useCallback(
       (timeRange: TimeRange) => {
         refUrlParams.current = {
           ...refUrlParams.current,
-          start: getParsedDate(timeRange.rangeFrom),
-          end: getParsedDate(timeRange.rangeTo, { roundUp: true }),
+          ...getDateRange({ state: {}, ...timeRange }),
```

Comment:
what about adding rangeId here? I don't think we need to expose incrementRangeId() then.